tests/client/api_v1/test_get_balance.c

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unity/unity.h>
+
+#include "client/api/v1/get_balance.h"
+#include "core/utils/byte_buffer.h"
+
+#define ADDR_HEX "7ed3d67fc7b619e72e588f51fef2379e43e6e9a856635843b3f29aa3a3f1f006"
+
+void test_get_balance() {
+  iota_client_conf_t conf = {
+      .url = "http://0.0.0.0/",
+      .port = 14265  // use default port number
+  };
+
+  res_balance_t* res = calloc(1, sizeof(res_balance_t));
```

Comment:
you need to free the response object after use.
dynamic allocation is not needed in this case, you can modify it like:
```suggestion
  res_balance_t res = {};
```

Reply:
I applied as you suggested: https://github.com/bernardoaraujor/iota.c/commit/52e94d4e5e95750afba3c39e595720fa11d71679

however that forces the `num == NULL` case on this conditional:
https://github.com/bernardoaraujor/iota.c/blob/52e94d4e5e95750afba3c39e595720fa11d71679/src/client/api/json_utils.c#L138

which leads to error here: https://github.com/bernardoaraujor/iota.c/blob/52e94d4e5e95750afba3c39e595720fa11d71679/src/client/api/v1/get_balance.c#L79